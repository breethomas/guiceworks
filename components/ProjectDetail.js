// @flow
import React from 'react'
import Markdown from 'react-markdown'
import Article from '../components/Article'
import ImageIcon from '../components/ImageIcon'
import Link from '../components/Link'
import PageSection from '../components/PageSection'
import Title from '../components/Title'
import type { ProjectProps } from '../types/app.js.flow'

const path = '/static/icons/projects'

type Props = {
  project: ProjectProps,
}

export default (props: Props) => (
  <PageSection>
    <Article>
      <ImageIcon
        alt={props.project.title}
        src={[path, props.project.icon].join('/')}
        width={250}
        height={90}
      />
      <Title>{props.project.title}</Title>
      <Link
        as={`/projects/${props.project.prev}`}
        href={`/projects?id=${props.project.prev}`}
      >
        { props.project.prev }
      </Link>
      <span> :: </span>
      <Link
        as={`/projects/${props.project.next}`}
        href={`/projects?id=${props.project.next}`}
      >
        { props.project.next }
      </Link>
      <Markdown source={props.project.body} />
    </Article>
  </PageSection>
)

