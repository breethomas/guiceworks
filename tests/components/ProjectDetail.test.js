import React from 'react'
import renderer from 'react-test-renderer'
import ProjectDetail from '../../components/ProjectDetail'

describe('ProjectDetail', () => {
  let component

  beforeEach(() => {
    const project = {
      body: 'Project 1 _body_',
      icon: 'icon.svg',
      id: 'project-1',
      image: '/static/images/image_home.jpg',
      greybox: '/static/images/image_grey-box.jpg',
      next: 'next-project',
      prev: 'prev-project',
      title: 'Project 1 title',
    }

    component = (
      <ProjectDetail
        project={project}
      />
    )
  })

  it('shows the correct snapshot for the ProjectDetail component tree', () => {
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
