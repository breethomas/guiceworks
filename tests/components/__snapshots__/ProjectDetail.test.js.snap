// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectDetail shows the correct snapshot for the ProjectDetail component tree 1`] = `
<section
  className={
    Object {
      "data-css-24spax": "",
    }
  }
>
  <header
    className={
      Object {
        "data-css-1f6b3ur": "",
      }
    }
  >
    <a
      className={
        Object {
          "data-css-33otqm": "",
        }
      }
      href="/projects/prev-project"
      onClick={[Function]}
      title="prev-project"
    >
      <svg
        aria-hidden={true}
        height={32}
        style={
          Object {
            "transform": "rotate(180deg)",
          }
        }
        viewBox="0 0 32 32"
        width={32}
      >
        <g
          fill="currentColor"
        >
          <polygon
            points="15.9 0 13.725 2.163 26.11 14.472 0 14.472 0 17.529 26.11 17.529 13.725 29.839 15.9 32 32 16"
          />
        </g>
      </svg>
    </a>
    <a
      className="css-33otqm css-omocl"
      href="/projects/next-project"
      onClick={[Function]}
      title="next-project"
    >
      <svg
        aria-hidden={true}
        height={32}
        viewBox="0 0 32 32"
        width={32}
      >
        <g
          fill="currentColor"
        >
          <polygon
            points="15.9 0 13.725 2.163 26.11 14.472 0 14.472 0 17.529 26.11 17.529 13.725 29.839 15.9 32 32 16"
          />
        </g>
      </svg>
    </a>
  </header>
  <article
    className="css-1d3ulb6 css-1jyyyen"
  >
    <header
      className={
        Object {
          "data-css-ub9vfp": "",
        }
      }
    >
      <img
        alt="Project 1 title"
        height={90}
        src="/static/icons/projects/icon.svg"
        width={250}
      />
    </header>
    <div
      className="css-js1aix css-uwpr5d css-jr10g3"
    >
      <p>
        Project 1 
        <em>
          body
        </em>
      </p>
    </div>
  </article>
</section>
`;
